### ISSUE TITLE
System.Text.Json roundtrips Stack<T> instances in reverse order

### ISSUE THREAD

**Issue by eiriktsarpalis** [dotnet/runtime#83086]
The code
```C#
var stack  = new Stack<int>(new int[] { 3, 2, 1 });

for (int i = 0; i < 5; i++)
{
    string json = JsonSerializer.Serialize(stack);
    Console.WriteLine(json);
    stack = JsonSerializer.Deserialize<Stack<int>>(json);
}
```
Produces
```
[1,2,3]
[3,2,1]
[1,2,3]
[3,2,1]
[1,2,3]
```
Even though this can be expected given the nature of stacks in general, it certainly violates expectations when it comes to marshalling collections. FWIW this is not specific to STJ as similar behavior can be observed when applying a stack to its own IEnumerable constructor.

**Comment by eiriktsarpalis** [comment]
FWIW I'm not necessarily proposing that we fix this (users might rely on the current behavior 😱)  although we should make sure that it is at least documented.

**Comment by gregsdennis** [comment]
My opinion is that if people are depending upon bug behavior, they shouldn't be surprised when that big is fixed, even if they don't know it's a bug.

I think fix it.

**Comment by netetireallyhuman** [comment]
I totally agree with [gregsdennis](https://github.com/gregsdennis). Please fix it.
